tools/RAiDER/models/ecmwf.py

```diff
@@ -101,11 +101,10 @@ def _load_model_level(self, fname):
         self._q = q
         geo_hgt, pres, hgt = self._calculategeoh(z, lnsp)
 
-        # re-assign lons, lats to match heights
-        _lons = np.broadcast_to(lons[np.newaxis, np.newaxis, :], hgt.shape)
-        _lats = np.broadcast_to(lats[np.newaxis, :, np.newaxis], hgt.shape)
+        self._lons, self._lats = np.meshgrid(lons, lats)
+
         # ys is latitude
-        self._get_heights(_lats, hgt)
+        self._get_heights(self._lats, hgt.transpose(1, 2, 0))
```

Comment:
this, I thought, should make the function work as internally in _get_heights we add a dimension to the last axis and then stack the lat/lons in height